### ISSUE TITLE
Default route gone after upgrade to 17.1

### ISSUE THREAD

**Issue by tkald** [opnsense/core#1362]
Upgrading Opnsense  from 16.7.14 to 17.1 loses default route on upgraded system. So basically no internet until you recheck "default gateway" on proper gateway.

Ha setup with 2 wan links.

**Comment by fichtner** [comment]
Hi,

It's very likely this commit from August 2016 https://github.com/opnsense/core/commit/ff568b4abfc7 that we kept on the 17.1 development track. In your case the semantics fit: no default was set but it was forced anyway.

The reason for removal was: explicit defaults, no magic for a "WAN" interface that e.g. wouldn't work in some Multi-WAN scenarios or when "WAN" as deleted for things like "OPT1" as seen on user's PPPoE setups. The removal had no impact so far.

Was this a clean OPNsense install or a pfSense import? Does this pose a problem still after setting the default gateways? 


Cheers,
Franco

**Comment by tkald** [comment]
It was upgrade from 16.7.14 console to 17.1. 
Since I upgraded Ha backup node first I noticed it lost outside connectivity. Cheked routing table and "default" route was missing. On pfsense gui, also no default gateway.

Prior master node upgrade I verified gui and routing table, default gateway/route was present. After upgrade to 17.1 it was gone.

It was clean Opnsense 16.7 install, no Pfsense involvment.
No further problems after setting one wan connection to default again.

But I think I may cause extra trouble for someone to figure out why their gateway(s) not working after upgrade.
